test/manipulation.test.js

```diff
@@ -382,7 +382,11 @@ describe('manipulation', function() {
             if (!err) {
               return done(new Error('Create should have rejected duplicate id.'));
             }
-            err.message.should.match(/duplicate/i);
+            if (db.adapter.name === 'ibmi') {
+              err.odbcErrors[0].message.should.match(/duplicate/i);
```

Comment:
What is `err.message` in this case? I think this test is intentionally checking `err.message` to ensure the framework is reporting consistent errors independently of the connector used. I would expect the connector to normalize the error object from ODBC style (in the case of `ibmi`) to LoopBack style.

Thoughts?

Reply:
`err.message` in this case would be something like `Error executing sql statement`. 

I know Sequelize has their own error class that is implemented by a number of sub classes (So `SequelizeError` is the parent class of `UniqueConstraintError` or `DuplicateKeyError`), and each connector intercepts any error output for the connector, parses the message/SQLState/code, and determines for itself which type of error it threw. That might be something to look into for the future.

For now, I'd say the vast majority of errors coming from the odbc connector contain just one error, so I could capture the error and just manually do something like:

```
// at the bottom of executeSQL... given a callback in invocation
this.pool.query(sql, params, (error, results) => {
  if (error) {
    error = error.odbcErrors[0];
  }
  callback(error, results);
});
```

Reply:
This works, will push changes to both PRs to remove the changes to this test.

Reply:
> For now, I'd say the vast majority of errors coming from the odbc connector contain just one error, so I could capture the error and just manually do something like:
> (...)
> This works, will push changes to both PRs to remove the changes to this test.

Excellent :+1:

Could you please verify that the stack traces reported for the inner errors (`error.odbcErrors[0]`) are offering enough information about the place where the error originated?

Reply:
I'm not sure that there is a good stack trace on those errors, unfortunately. The errors are created in N-API, and it doesn't look like I have ensured a stack track is created. I have modified the connector so that it creates a stack trace to the pool.query call, but below that there isn't a good trace. Have added it to my TODOs for the `odbc` package.